List/Circular Linked List/list_test_cases.cpp

```diff
@@ -0,0 +1,385 @@
+#include "List.hpp"
+#include <cassert>
+#include <iostream>
+#include <sstream>
+
+using namespace std;
+
+void assert_notice(std::string name, bool passed) {
+  std::cout << "\n" << name << " " << (passed ? "passed" : "failed") << ".\n";
+}
+
+void test_list_constructor() {
+  List list;
+  assert(list.length() == 0);
+  assert_notice(__func__, true);
```

Comment:
Please describe what is `__func__` for?
Provide comment at the start of the file to describe what is `__func__` and what it is used for.